### PR TITLE
Feature/route test setup

### PR DESCRIPTION
**Pull request by govTechSatish**

### CODE REVIEW

File: test/setup.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  await server.initialize()
+  global.__SERVER__ = server
+  global.__VALIDSESSION__ = true
+  global.__AUTH__ = { credentials: { reference: '1111', sbi: '111111111' }, strategy: 'basic' }
+  const crumb = await getCrumbs(server)
+  global.__CRUMB_VALUE__ = crumb
+  global.__CRUMB_HEADER__ = { cookie: `crumb=${crumb}` }
+})
+const getCrumbs = require('../test/utils/get-crumbs')
+describe('Login test', () => {
```
Comment (st3v3nhunt) [inline test/setup.js:15]: Should this test be in the setup or its own test file?

**Review by st3v3nhunt — COMMENTED** [review]
I like the approach of using the global setup and tear down files for the `server` as it removes the boiler plate code for doing this from the individual tests and that is pretty much the same in every test file.

Regarding the other values i.e. `VALIDSESSION`, `AUTH`, `CRUMB`, etc. I think I would prefer they were setup in the individual test files so the details of them are easily seen alongside the tests. Plus, there are going to be tests that need different setup for crumbs and session validity, etc. What do you think?

File: test/setup.js (hunk 1 above)
Comment (govTechSatish) [reply test/setup.js:15]: thats right. Thanks.

File: test/integration/narrow/routes/farmer-apply/sheep.test.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    const crumb = await getCrumbs(global.__SERVER__)
+    const options = {
+      method: 'POST',
+      url: '/farmer-apply/sheep',
+      payload: { crumb, sheep: null },
+      auth,
+      headers: { cookie: `crumb=${crumb}` }
+    }
+    const res = await global.__SERVER__.inject(options)
+    expect(res.payload).toContain('Select yes if you keep more than 20 sheep')
```
Comment (st3v3nhunt) [inline test/integration/narrow/routes/farmer-apply/sheep.test.js:39]: See the comment regarding using `cheerio` to enable specific content to be selected from the page.

**Review by st3v3nhunt — CHANGES_REQUESTED** [review]
Looks good, a couple of comments. The first one relates to a couple of blank lines.

And the second one which is applicable to all of the tests - rather than checking `res.payload` contains some string, if you load `res.payload` into `cheerio` (example in comments) you can select specific elements of the response to test against which is useful as it means changes in the view should be picked up. e.g. you can select the `h1` or a button or an input on a form, etc.

**Review by st3v3nhunt — APPROVED** [review]
LGTM 👍

**Pull request merged**